### PR TITLE
add menu LaTeX/Horizontal Spacing

### LINKED ISSUE #2687: Horizontal Spacing commands in the menu for common commands
It would be nice to have common horizontal spacing commands in the LaTeX-Command-Dropdown.

### Who will benefit with this feature?
Everybody who wants to add horizontal spacing? (especially me)

### Any Other info
The menu already includes the commands for vertical spacing.

### PR DESCRIPTION
**Pull request by octaeder**
This PR closes #2687.

Additionally:
- added \vfill to menu Vertical Spacing

Question: Where are the info texts displayed or are they only documentation? Hoped they would be tooltips.

### CODE REVIEW

**Comment by mbertucci47** [comment]
I would not include `\baselineskip` since it is a length/skip, not a document command like the rest in that list. Unrelated, the words "positive" and "negative" in the `\vspace` entry are missing the "e", and "vertically" in the `\vfill` entry has an extra "t".

File: src/texstudio.cpp [hunk 1]
```diff
@@ -1206,7 +1206,7 @@ void Texstudio::setupMenus()
 	newManagedAction(submenu, "remHLine", tr("Remove \\hline", "table"), SLOT(remHLineCB()));
 	newManagedAction(submenu, "insertTableTemplate", tr("Remodel Table Using Template", "table"), SLOT(insertTableTemplate()));
 	newManagedAction(submenu, "alignColumns", tr("Align Columns"), SLOT(alignTableCols()), QKeySequence(), "alignCols");
-	submenu = newManagedMenu(menu, "magicComments", tr("Add magic comments ..."));
+	submenu = newManagedMenu(menu, "magicComments", tr("Add magic comments"));
```
Comment (sunderme) [inline src/texstudio.cpp:1209]: there is no real benefit for this change, but all translations are buggered. So, I prefer to not change it.

File: uiconfig.xml [hunk 2]
```diff
@@ -12,21 +12,21 @@
...
+    <insert id="subparagraph" text="\subparagraph" insert="\subparagraph{%&lt;title%>}"/>
 
-    <insert id="part*" text="part*" insert="\part*{%&lt;title%>}"/>
-    <insert id="chapter*" text="chapter*" insert="\chapter*{%&lt;title%>}"/>
-    <insert id="section*" text="section*" insert="\section*{%&lt;title%>}"/>
-    <insert id="subsection*" text="subsection*" insert="\subsection*{%&lt;title%>}"/>
-    <insert id="subsubsection*" text="subsubsection*" insert="\subsubsection*{%&lt;title%>}"/>
-    <insert id="paragraph*" text="paragraph*" insert="\paragraph*{%&lt;title%>}"/>
-    <insert id="subparagraph*" text="subparagraph*" insert="\subparagraph*{%&lt;title%>}"/>
+    <insert id="part*" text="\part*" insert="\part*{%&lt;title%>}"/>
```
Comment (sunderme) [inline uiconfig.xml:23]: this and following.
txs has used this scheme for 15(?) years now. There absolutely no need to change it except it messes up translations. (and it looks nicer without the backslash)

File: uiconfig.xml [hunk 3]
```diff
@@ -92,18 +92,18 @@
...
-     <insert id="makebox" text="makebox" insert="\makebox[][]{%&lt;content%>}"/>
-     <insert id="fbox" text="fbox" insert="\fbox{%&lt;content%>}"/>
-     <insert id="framebox" text="framebox" insert="\framebox[][]{%&lt;content%>}"/>
-     <insert id="newsavebox" text="newsavebox" insert="\newsavebox{%&lt;name%>}"/>
-     <insert id="sbox" text="sbox" insert="\sbox{%&lt;name%>}{%&lt;content%>}"/>
-     <insert id="savebox" text="savebox" insert="\savebox{%&lt;name%>}[][]{%&lt;content%>}"/>
-     <insert id="usebox" text="usebox" insert="\usebox{%&lt;name%>}"/>
-     <insert id="raisebox" text="raisebox" insert="\raisebox{%&lt;voffset%>}[][]{%&lt;content%>}"/>
-     <insert id="parbox" text="parbox" insert="\parbox[]{%&lt;width%>}{%&lt;content%>}"/>
+     <insert id="mbox" text="\mbox" insert="\mbox{%&lt;content%>}"/>
```
Comment (sunderme) [inline uiconfig.xml:95]: same as above

**Comment by sunderme** [comment]
I would **strongly** recommend not to mix different changes in one PR.
One PR for adding a part of the menu which probably would be accepted quite easily and other PRs for renaming stuff (which are not accepted as stated above)
I can not cherry-pick ...

**Comment by octaeder** [comment]
understood
reason was that the menus are a mix of commands with slashes and without slashes
menu item with three dots means that a dialog will pop up, so selecting the item can still be canceled (windows style guide?)
thx for reviewing to @mbertucci47 and @sunderme

File: uiconfig.xml [hunk 4]
```diff
@@ -142,14 +142,29 @@
...
+
+  <menu id="verticalSpacing" text="&amp;Vertical Spacing">
     <insert id="newpage" text="\newpage" insert="\newpage" info="The \newpage command ends the current page"/>
     <insert id="linebreak" text="\linebreak" insert="\linebreak" info="The \linebreak command tells LaTeX to break the current line at the point of the command."/>
     <insert id="pagebreak" text="\pagebreak" insert="\pagebreak" info="The \pagebreak command tells LaTeX to break the current page at the point of the command."/>
     <insert id="bigskip" text="\bigskip" insert="\bigskip" info="The \bigskip command adds a 'big' vertical space."/>
     <insert id="medskip" text="\medskip" insert="\medskip" info="The \medskip command adds a 'medium' vertical space."/>
     <insert id="smallskip" text="\smallskip" insert="\smallskip" info="The \smallskip command adds a 'small' vertical space."/>
-    <insert id="vspace" text="\vspace" insert="\vspace{%&lt;skip%>}"/>
+    <insert id="baselineskip" text="\baselineskip" insert="\baselineskip" info="The \baselineskip command adds the normal distance between lines in a paragraph."/>
```
[the remark this answers is not in the document]
Comment (octaeder) [reply uiconfig.xml:165]: txh for clarifying

**Comment by sunderme** [comment]
thanks

**Pull request merged**